h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -98,7 +98,7 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
       Frame specTrainFrame = _hyperSpaceWalker.getParams().train();
       Frame oldTrainFrame = grid.getTrainingFrame();
       if (oldTrainFrame != null && !specTrainFrame._key.equals(oldTrainFrame._key) ||
```

Comment:
super minor idea: We could check for null as part of the current condition just by encapsulating the second statement in the condition by parentheses - ```if(oldTrainFrame != null && (!specTrainFrame._key.equals(oldTrainFrame._key) || specTrainFrame.checksum() != oldTrainFrame.checksum())```